controllers/actions.summerwind.net/horizontal_runner_autoscaler_batch_scale.go

```diff
@@ -169,12 +169,48 @@ func (s *batchScaler) batchScale(ctx context.Context, batch batchScaleOperation)
 		} else if amount < 0 {
 			var reservations []v1alpha1.CapacityReservation
 
-			var found bool
+			var (
+				found    bool
+				foundIdx int
+			)
 
-			for _, r := range copy.Spec.CapacityReservations {
+			for i, r := range copy.Spec.CapacityReservations {
+				r := r
 				if !found && r.Replicas+amount == 0 {
 					found = true
+					foundIdx = i
 				} else {
+					// Note that we nil-check max replicas because this "fix" is needed only when there is the upper limit of runners.
```

Comment:
>is needed only when there is the upper limit of runners.

As a work around we tried to use only metrics scaling with PercentageRunnersBusy. This bug hits again where the effective time is not updated so we have to wait 10mins between scale ups. Even when not at our max, and not using webhook scaling.
https://github.com/actions/actions-runner-controller/blob/b5e9e1424497d6b4886ac94c2b6a8f6a5866a239/controllers/actions.summerwind.net/runner_pod_owner.go#L336

Reply:
Yeah, so if you start from a new HRA for pull-based scaling, it would just work! However note that pull-based scaling is deprecated and known to have several issues. The biggest would be that pull-based scaling takes time to trigger... (approx 60s since the runner becomes busy/worfklow job gets queued due to github API cache)

Reply:
All in all, I think we should merge this and keep recommending the use of webhook-based scaling!

Reply:
Good to know. I'm happy if you're happy. If you have time to cut a patch release I can test more tomorrow.